lib/Sema/SemaAPINotes.cpp

```diff
@@ -149,8 +171,14 @@ namespace {
       auto end = D->specific_attr_end<A>();
       auto existing = getExistingAttr(D);
       if (existing != end) {
-        // Remove the existing attribute.
+        // Remove the existing attribute, and treat it as a superseded
+        // non-versioned attribute.
+        auto *versioned =
+            SwiftVersionedAttr::CreateImplicit(S.Context, clang::VersionTuple(),
+                                               *existing);
+
         D->getAttrs().erase(existing.getCurrent());
+        D->addAttr(versioned);
```

Comment:
I guess an empty `VersionTuple` is code for "latest version". That seems fine to me, and this looks like the right way to maintain the attribute that was present in the source code without the rest of Clang (or other tools that only care about the API-note-adjusted view) seeing it.